x-pack/plugins/uptime/public/lib/alert_types/monitor_status.tsx

```diff
@@ -10,27 +10,64 @@ import { isRight } from 'fp-ts/lib/Either';
 import { PathReporter } from 'io-ts/lib/PathReporter';
 import { AlertTypeModel } from '../../../../triggers_actions_ui/public';
 import { AlertTypeInitializer } from '.';
-import { AtomicStatusCheckParamsType, StatusCheckParamsType } from '../../../common/runtime_types';
+import {
+  AtomicStatusCheckParamsType,
+  StatusCheckParamsType,
+  MonitorAvailabilityType,
+} from '../../../common/runtime_types';
 import { MonitorStatusTitle } from './monitor_status_title';
 import { CLIENT_ALERT_TYPES } from '../../../common/constants';
 import { MonitorStatusTranslations } from './translations';
 import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
 import { store } from '../../state';
 
+interface HasShouldCheckFlags {
+  shouldCheckAvailability?: boolean;
+  shouldCheckStatus?: boolean;
+}
+
+/**
+ * The alert parameters are of `unknown` type, so we must determine if those fields exist.
+ * If either field is missing, this function will return false.
+ * @param value The alert parameters value.
+ */
+function hasShouldCheckFlags(value: unknown): value is HasShouldCheckFlags {
+  const types = ['boolean', 'undefined'];
```

Comment:
It's neat using the `unknown` type in this way, and I'm not going to say we should change anything anything here, but I guess I don't see the benefit of the extra ceremony casting with `as` etc. We could also just use `any` and check the property and have identical behavior. The question is whether we see the problem as "Let's figure out what type alert data is then determine what it is strongly" vs. "An alert is a bag of data, let's see if it has this property". Neither way is right, but one has less ceremony. Would be curious to hear other thoughts on this though.

Reply:
We don't need to use `unknown` here. For some reason I thought it was required by the alerting API, but it's not. The type they expect is effectively `any`, so in this case I'm fine with switching the type to `any` and removing this extra code. https://github.com/elastic/kibana/pull/70284/commits/3a109af4649b042a2b471e9a582bb50b5c4408ea